src/error.rs

```diff
@@ -69,6 +69,13 @@ impl From<IoError> for Error {
     }
 }
 
+/// Special case for `inflate_bytes` where the error is a String
+impl From<String> for Error {
+    fn from(msg: String) -> Self {
+        Error::Invalid(msg.into())
+    }
+}
+
```

Comment:
I think, in the future, someone could be surprised to find out that all Strings are implicitly converted to `Invalid`, but never to `Unsupported` or the like.  I suppose it would be more "obvious" to avoid this implementation and instead use `compressor.finish().map_err(|string| Error::invalid(string))` inside `zip.rs`. What do you think?

Reply:
I have to agree, this can bring more trouble in the future than it should.